ecommerce_integrations/shopify/order.py

```diff
@@ -156,20 +156,35 @@ def get_order_items(order_items, setting, delivery_date):
 
 def get_order_taxes(shopify_order, setting):
 	taxes = []
-	for tax in shopify_order.get("tax_lines"):
-		taxes.append(
-			{
-				"charge_type": _("On Net Total"),
-				"account_head": get_tax_account_head(tax),
-				"description": f"{get_tax_account_description(tax) or tax.get('title')} - {tax.get('rate') * 100.0:.2f}%",
-				"rate": tax.get("rate") * 100.00,
-				"included_in_print_rate": 1 if shopify_order.get("taxes_included") else 0,
-				"cost_center": setting.cost_center,
-			}
-		)
-
+	if not setting.dont_recalculate_taxes:
+		for tax in shopify_order.get("tax_lines"):
+			taxes.append(
+				{
+					"charge_type": _("On Net Total"),
+					"account_head": get_tax_account_head(tax),
+					"description": f"{get_tax_account_description(tax) or tax.get('title')} - {tax.get('rate') * 100.0:.2f}%",
+					"rate": tax.get("rate") * 100.00,
+					"included_in_print_rate": 1 if shopify_order.get("taxes_included") else 0,
+					"cost_center": setting.cost_center,
+				}
+			)
+		
 	taxes = update_taxes_with_shipping_lines(taxes, shopify_order.get("shipping_lines"), setting)
 
+	if setting.dont_recalculate_taxes:
+		for tax in shopify_order.get("tax_lines"):
+			print(tax)
+			taxes.append(
+				{
+					"charge_type": _("Actual"),
+					"account_head": get_tax_account_head(tax),
+					"description": f"{tax.get('title')} - {tax.get('rate') * 100.0}%",
+					"tax_amount": tax.get("price"),
+					"included_in_print_rate": 1 if shopify_order.get("taxes_included") else 0,
```

Comment:
"Actual" and "tax included" can't work together.


IMO we should just be using the "Actual" type, might have to subtract the taxes from basic rates if rates include taxes. Then only we can use the "Actual" charge type. 

This is a rather complicated requirement and will probably require modifying the way ERPNext does taxes which is okay for users but not so much for integrations.

Reply:
I like your idea of substracting it from basic rate, but there is a problem. We need to know if shipping charge contains taxes or not. If yes the actual tax splits up in the part of basic rate of item and shipping... you have an idea for that?